packages/edge-config/src/index.ts

```diff
@@ -127,6 +127,22 @@ function getConnection(connectionString: string): Connection | null {
   }
 }
 
+interface EdgeConfigClientOptions {
+  /**
+   * The stale-if-error response directive indicates that the cache can reuse a
+   * stale response when an upstream server generates an error, or when the error
+   * is generated locally - for example due to a connection error.
+   *
+   * Any response with a status code of 500, 502, 503, or 504 is considered an error.
+   *
+   * Pass `Infinity` to always use stale cached entries on error.
+   * Pass a negative number to turn disable stale-if-error semantics.
+   *
+   * The time is supplied in seconds. Defaults to `Infinity`.
+   */
+  staleIfError: number;
```

Comment:
I made `staleIfError` configurable and pass it down to `fetchWithCachedResponse`, but I wonder if we should instead pass it using a request header directive. This would work perfectly for this case, but might not work for other kinds of options. I'm still going back and forth, but leaning towards passing it using a request header instead.